### PR TITLE
Document and advertise Deploy to OpenShift

### LINKED ISSUE #359: Let users know they can deploy to openshift
Once the new deploy to openshift command is available (#316),  we should promote the new feature 
- in the welcome page: https://github.com/redhat-developer/vscode-quarkus/blob/master/assets/webviews/templates/welcome.ejs
- by showing a popup (only once!), advertising the feature, once the user execute the debug command

### PR DESCRIPTION
**Pull request by datho7561**
Closes #359

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

**Comment by datho7561** [comment]
~~I need to document that we emit telemetry events when the Deploy to OpenShift is run!!~~ Done

File: USAGE_DATA.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ vscode-quarkus has opt-in telemetry collection, provided by [vscode-commons](htt
    The following commands emit these telemetry events:
     * "Quarkus: Add extensions to current project"
     * "Quarkus: Debug current Quarkus project"
+    * "Quarkus: Deploy current Quarkus Project to OpenShift"
```
Comment (rgrunber) [inline USAGE_DATA.md]: I'd probably update this to reflect the changes made in https://github.com/redhat-developer/vscode-quarkus/pull/374 . Here and in the README.

File: src/wizards/debugging/startDebugging.ts [hunk 2]
```diff
@@ -48,3 +54,16 @@ export async function startDebugging(): Promise<void> {
...
+
+const DEPLOY_TO_OPENSHIFT_ADVERTISEMENT_MEMENTO = 'deployToOpenshift.advertisementShown';
+
+function hasShownDeployToOpenShiftPopUp(): boolean {
+  return QuarkusContext.getExtensionContext().globalState.get(DEPLOY_TO_OPENSHIFT_ADVERTISEMENT_MEMENTO, false);
+}
+
+function showDeployToOpenShiftPopUp(): void {
+  const DEPLOY_TO_OPENSHIFT_ADVERTISEMENT = 'You can deploy to OpenShift right from VS Code using ' + //
+      '"Quarkus: Deploy Current Project ot OpenShift" from the command palette';
```
Comment (rgrunber) [inline src/wizards/debugging/startDebugging.ts]: Update to match based on https://github.com/redhat-developer/vscode-quarkus/pull/374 .

File: assets/webviews/templates/welcome.ejs [hunk 3]
```diff
@@ -35,10 +35,27 @@
...
+      <div class="section-div">
+        <h2 class="section-title">Get started</h2>
+        <p>Create a new Quarkus project or open an existing Quarkus project to get started.</p>
+        <a href="command:quarkusTools.createProject"><button>Create Quarkus project</button></a>
+      </div>
+    </div>
+    <div class="column">
+      <div class="section-div">
+        <h2 class="section-title">
+          Notable Features
```
Comment (rgrunber) [inline assets/webviews/templates/welcome.ejs:49]: Is `quarkusTools.buildBinary` worth mentioning ?

File: assets/webviews/templates/welcome.ejs [hunk 4]
```diff
@@ -35,10 +35,27 @@
...
+        </h2>
+        <ul>
+          <li>Intellisense for property keys and values in Quarkus <code>.properties</code> files</li>
```
Comment (rgrunber) [inline assets/webviews/templates/welcome.ejs:52]: Is it worth mentioning the additional support for Java files ?

**Review by rgrunber — APPROVED** [review]
LGTM. I was wondering why the command string isn't shown as a link, but I guess if the user has already launched a debug session, it doesn't make sense to give them the option to immediately launch a deploy command.

**Comment by datho7561** [comment]
I could take a look into providing links to run the commands from the Welcome page. Do you think it would be worthwhile?

**Comment by rgrunber** [comment]
> I could take a look into providing links to run the commands from the Welcome page. Do you think it would be worthwhile?

I think it's fine as-is for now. I meant in the notification popup (one-time), the command is show as a string, but I guess this is fine because form within that context, it probably wouldn't make sense to give users the option to launch again.

**Pull request merged**